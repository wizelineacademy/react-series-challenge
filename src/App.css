.app {
  display: block;
}

.main {
  padding: 2rem;
}

.navigation {
  padding: 2rem 0 0 0;
}

.link:any-link {
  color: #B0B0B5;
  margin: 0.25rem 0rem;
  padding: 0.5rem 1rem 0.5rem;
  text-decoration: none;
  transition-property: all;
  transition-duration: 400ms;
  transition-timing-function: ease-in-out
}

.link.active {

}

@media screen and (min-width: 32rem) {
  .app {
    display: grid;
    grid-template-areas:  "header header"
                          "aside main";
    grid-template-columns: 2fr 10fr;
  }

  .aside {
    grid-area: aside;
  }

  .main {
    grid-area: main;
  }

  .header {
    grid-area: header;
  }

  .navigation {
    display: flex;
    flex-direction: column;
  }
}