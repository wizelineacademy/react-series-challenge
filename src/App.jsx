import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import store from './store';
import AppRouting from './AppRouting';

const App = () => (
  <Provider store={store}>
    <Router>
      <AppRouting />
    </Router>
  </Provider>
);

export default App;
