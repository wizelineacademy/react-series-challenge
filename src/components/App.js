import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import HomePage from "./pages/HomePage/index";
import FavsPage from "./pages/FavsPage/index";

const App = () => {
  return (
    <Switch>
      <Route exact path="/" render={() => <Redirect to="/home" />} />
      <Route path="/home" component={HomePage} />;
      <Route path="/favorites" component={FavsPage} />;
    </Switch>
  );
};

export default App;
