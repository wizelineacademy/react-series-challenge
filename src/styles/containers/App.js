import styled from 'styled-components';

const CenterContenteDiv = styled.div`
  text-align: center;
  width: 100%;
  height: 100%;

  background: linear-gradient(black, white);
`;

export default CenterContenteDiv;
