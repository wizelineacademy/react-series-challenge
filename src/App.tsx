import * as React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import { Switch, Route, Redirect } from 'react-router-dom';
import { Provider} from 'react-redux';

import store from './store';
import ROUTES from './routes';
import Header from './components/presentational/Header';
import { Container } from './components/presentational/Layout';

import './App.style'

class App extends React.Component {
  public render() {
    return (
      <Provider store={store}>
        <Router>
          <Container>
            <Header />
            <section>
            <Switch>
              {ROUTES.map(({path, component, exact}) =>
                <Route key={path} path={path} exact={exact} component={component} />)}
              <Redirect to='/home' />
            </Switch>
            </section>
          </Container>
        </Router>
      </Provider>
    );
  }
}
export default App;
