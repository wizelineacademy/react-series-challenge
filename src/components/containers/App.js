import React, { Component } from 'react';
import {Route, Switch} from 'react-router-dom';
import {Menu, ItemMenu} from '../presentational';
import {AppContainer} from './App.styled';
import Home from './Home';
import Favorites from './Favorites';
import GifDetail from './GifDetail';
import NotFound from './NotFound';

class App extends Component {
  render() {
    return (
      <AppContainer>
        <header>
          <Menu>
            <ItemMenu exact to='/'>Home</ItemMenu>
            <ItemMenu to='/favorites'>Favorites</ItemMenu>
          </Menu>
        </header>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/favorites" component={Favorites} />
          <Route 
            path="/gifs/:gif_id" 
            component={({match}) => {
              return <GifDetail gifID={match.params.gif_id} />
            }} />
          <Route component={NotFound} />
        </Switch>
      </AppContainer>
    );
  }
}

export default App;
