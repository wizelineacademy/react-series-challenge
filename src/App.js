import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import styled from 'styled-components';
import Header from './components/presentational/Header';
import Menu from './components/presentational/Menu';
import { TrendingGiphys, Favorites, GiphyDetail } from './components/pages';

const sections = [
  { route: 'trending-giphy', title: 'Trending' , inMenu: true, component: TrendingGiphys },
  { route: 'favorites', title: 'My favorites' , inMenu: true, component: Favorites },
  { route: 'giphy/:giphyId', title: 'Giphy Detail' , inMenu: false, component: GiphyDetail },
];

const Wrapper = styled.div`
    display: flex;  
  `;

class App extends Component {
  renderRoutes(sections) {
    return sections.map(section => {
      return (
        <Route path={`/${section.route}`} component={section.component} key={section.route} />
      );
    });
  }


  render() {
    return (
      <div>
        <Header />
        <Wrapper>
          <Menu items={sections.filter(section => section.inMenu)} />
          <Switch>
            {this.renderRoutes(sections)}
          </Switch>
        </Wrapper>
      </div>
    );
  }
}

export default App;
