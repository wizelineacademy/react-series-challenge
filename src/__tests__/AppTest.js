import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'

import App from '../App';

describe('App', () => {
  it('should render without crashing', () => {
    const testApp = shallow( <App />);    

    expect(shallowToJson(testApp)).toMatchSnapshot()
  })
})




