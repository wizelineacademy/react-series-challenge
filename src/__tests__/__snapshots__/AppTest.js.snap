// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div>
  <Header />
  <styled.div>
    <Menu
      items={
        Array [
          Object {
            "component": [Function],
            "inMenu": true,
            "route": "trending-giphy",
            "title": "Trending",
          },
          Object {
            "component": [Function],
            "inMenu": true,
            "route": "favorites",
            "title": "My favorites",
          },
        ]
      }
    />
    <Switch>
      <Route
        component={[Function]}
        key="trending-giphy"
        path="/trending-giphy"
      />
      <Route
        component={[Function]}
        key="favorites"
        path="/favorites"
      />
      <Route
        component={[Function]}
        key="giphy/:giphyId"
        path="/giphy/:giphyId"
      />
    </Switch>
  </styled.div>
</div>
`;
